pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -306,6 +307,21 @@ public long getLatestIngestionTimestamp() {
       MutableInvertedIndex invertedIndexReader =
           invertedIndexColumns.contains(column) ? indexProvider.newInvertedIndex(context.forInvertedIndex()) : null;
 
+      MutableTextIndex fstIndex = null;
+      //FST Index
+      if (fstIndexColumns.contains(column)) {
+        if (_fieldConfigList != null) {
```

Comment:
nit: you can combine these nested if statement into one like:
 `if (_fieldConfigList != null && fstIndexColumns.contains(column))`